### PR TITLE
Activation code for the third hard fork

### PR DESCRIPTION
**Pull request by dagurval**
On top of #374

### CODE REVIEW

**Comment by dgenr8** [comment]
Needs rebase

**Comment by dgenr8** [comment]
Now that the third HF time is well established, this PR can include a reversion of https://github.com/bitcoinxt/bitcoinxt/commit/c77e6fbc1cbfd0fd3d7501b4be80a467c8a77ad0.

**Comment by dagurval** [comment]
rebased and reverted commit c77e6fb as @dgenr8 suggested

File: src/main.cpp [hunk 1]
```diff
@@ -1234,8 +1234,11 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
             return state.DoS(0, false, REJECT_NONSTANDARD, "too-long-mempool-chain", false);
         }
 
-        unsigned int forkVerifyFlags = ((Opt().UAHFTime() != 0) && (chainActive.Tip()->GetMedianTimePast() >= Opt().UAHFTime()) ?
+        const int64_t mtpChainTip = chainActive.Tip()->GetMedianTimePast();
+        unsigned int forkVerifyFlags = ((Opt().UAHFTime() != 0) && (mtpChainTip >= Opt().UAHFTime()) ?
```
Comment (dagurval) [inline src/main.cpp]: This could use `IsUAHFActive` from utilfork instead

Comment (dgenr8) [reply src/main.cpp]: Sounds great

File: src/utilfork.cpp [hunk 2]
```diff
@@ -31,8 +31,20 @@ bool IsThirdHFActivatingBlock(int64_t mtpCurrent, const CBlockIndex* pindexPrev)
...
+bool IsUAHFActive(uint64_t mtpChainTip) {
+    return IsForkActive(Opt().UAHFTime(), mtpChainTip);
+}
+
 bool IsThirdHFActive(int64_t mtpChainTip) {
-    int64_t mtpHF = Opt().ThirdHFTime();
+    return IsForkActive(Opt().ThirdHFTime(), mtpChainTip);
+}
 
     if (!mtpHF)
```
Comment (dgenr8) [inline src/utilfork.cpp]: Space junk here.

File: src/utilfork.cpp [hunk 3]
```diff
@@ -46,8 +48,40 @@ bool IsThirdHFActive(int64_t mtpChainTip) {
     return IsForkActive(Opt().ThirdHFTime(), mtpChainTip);
 }
 
-    if (!mtpHF)
-        return false;
+void ForkMempoolClearer(
+        CTxMemPool& mempool, const CBlockIndex* oldTip, const CBlockIndex* newTip)
+{
+    if (oldTip == nullptr || newTip == nullptr)
```
Comment (dgenr8) [inline src/utilfork.cpp]: `|| oldTip == newTip`

**Comment by dagurval** [comment]
Comments addressed @dgenr8

**Comment by dagurval** [comment]
@xt-bot retest this

**Comment by dgenr8** [comment]
ACK

**Pull request merged**